rust/blockstore/src/arrow/block/value/int32array_value.rs

```diff
@@ -43,6 +52,38 @@ impl ArrowWriteableValue for Vec<u32> {
     fn get_delta_builder() -> BlockStorage {
         BlockStorage::VecUInt32(SingleColumnStorage::new())
     }
+
+    fn get_arrow_builder(capacity_hint: Self::ArrowCapacityHint) -> Self::ArrowBuilder {
+        let total_value_count = capacity_hint.byte_size / size_of::<u32>();
+        ListBuilder::with_capacity(
+            UInt32Builder::with_capacity(total_value_count),
+            capacity_hint.item_count,
+        )
+    }
+
+    fn prepare(value: Self) -> Self::PreparedValue {
```

Comment:
To short circuit figuring it out myself, what is a "Prepared Value"

Reply:
https://github.com/chroma-core/chroma/pull/2961#discussion_r1802128422

> Currently, prepare() primarily simplifies implementation of DataRecord writing as the transient in-memory delta representation differs from the struct, but in the future it could also potentially improve efficiency by avoiding serializing multiple times to calculate the size.